File: Algorithm/programmers/고득점키트/dfsbfs/타겟넘버.py
def go(now, numbers, target, idx):
    global answer

    if idx == len(numbers):
        if now == target:
            answer += 1
        return

    go(now + numbers[idx], numbers, target, idx + 1)
    go(now - numbers[idx], numbers, target, idx+1)


answer = 0


def solution(numbers, target):
    go(0, numbers, target, 0)
    return answer

File: Algorithm/programmers/고득점키트/dfsbfs/타겟넘버.js
let answer = 0;

const go = (now, numbers, target, idx) => {
  if (idx == numbers.length) {
    if (now === target) {
      console.log(now, target, answer)
      answer += 1};
    return;
  }

  go(now + numbers[idx], numbers, target, idx + 1);
  go(now - numbers[idx], numbers, target, idx + 1);
};

const solution = (numbers, target) => {
  go(0, numbers, target, 0);
  return answer;
};

const numbers = [1,1,1,1,1]
target = 3
console.log(solution(numbers, target))